capstone-frontend/src/components/ClubPage.jsx

```diff
@@ -17,59 +17,61 @@ class ClubPage extends Component {
     let memberArray = this.state.members;
     let memberIDsArray = this.state.members.map(m => m.id);
     const index = memberIDsArray.indexOf(memberID);
     if (index > -1) {
       memberArray.splice(index, 1);
     }
-    this.setState({members: memberArray});
+    this.setState({ members: memberArray });
   }
 
   fetchData = async () => {
     await fetch(`/api/clubs?id=${this.props.id}`)
-        .then(response => response.json()).then(clubJson => this.setState({club: clubJson[0]}))
-        .catch(function(err) {
-            //TODO #61: Centralize error output
-            alert(err);
-        });
+      .then(response => response.json()).then(clubJson => this.setState({ club: clubJson[0] }))
+      .catch(function (err) {
+        //TODO #61: Centralize error output
+        alert(err);
+      });
 
-    await fetch(`/api/search?gbookId=${this.state.club.gbookID}`)
-        .then(response => response.json()).then(bookJson => this.setState({book: bookJson[0]}))
-        .catch(function(err) {
-            //TODO #61: Centralize error output
-            alert(err);
+    if (this.state.club.gbookID.length > 0) {
+      await fetch(`/api/search?gbookId=${this.state.club.gbookID}`)
+        .then(response => response.json()).then(bookJson => this.setState({ book: bookJson[0] }))
+        .catch(function (err) {
+          //TODO #61: Centralize error output
+          alert(err);
         });
+    }
 
     await fetch(`/api/assignments?clubID=${this.state.club.id}`)
-        .then(response => response.json()).then(assignmentJson => this.setState({assignments: assignmentJson}))
-        .catch(function(err) {
-            //TODO #61: Centralize error output
-            alert(err); 
-        });
+      .then(response => response.json()).then(assignmentJson => this.setState({ assignments: assignmentJson }))
+      .catch(function (err) {
+        //TODO #61: Centralize error output
+        alert(err);
+      });
 
     await fetch(`/api/user?id=${this.state.club.ownerID}`)
-        .then(response => response.json()).then(ownerJson => this.setState({owner: ownerJson}))
-        .catch(function(err) {
-            //TODO #61: Centralize error output
-            alert(err); 
-        });
-        
-    this.setState({members: []});
+      .then(response => response.json()).then(ownerJson => this.setState({ owner: ownerJson }))
+      .catch(function (err) {
+        //TODO #61: Centralize error output
+        alert(err);
+      });
+
+    this.setState({ members: [] });
     for (let i = 0; i < this.state.club.memberIDs.length; i++) {
       await fetch(`/api/user?id=${this.state.club.memberIDs[i]}`)
-          .then(response => response.json())
-          .then(memberJson => memberJson && this.setState({members: [...this.state.members, memberJson]}))
-          .catch(function(err) {
-            //TODO #61: Centralize error output
-            alert(err); 
-          });
+        .then(response => response.json())
```

Comment:
I know all this formatting is done by Gitpod, but it may create merge conflicts later. Changing unrelated files makes PRs more difficult to review, but I know this styling is automated.

Reply:
+1

Reply:
Yeah sorry, I didn't mean to format this file.